lib/internal/child_process.js

```diff
@@ -182,30 +180,30 @@ function ChildProcess() {
     //
     // - spawn failures are reported with exitCode < 0
     //
-    var syscall = self.spawnfile ? 'spawn ' + self.spawnfile : 'spawn';
+    var syscall = this.spawnfile ? 'spawn ' + this.spawnfile : 'spawn';
     var err = (exitCode < 0) ? errnoException(exitCode, syscall) : null;
```

Comment:
Does this and the other `exitCode < 0` check below (line 201 or 199 depending on whether you're looking at it before the change or after) make [the `code < 0` check in `lib/child_process.js`](https://github.com/nodejs/node/blob/b8f416023d61c73c571f15bc967d6ec630e3215f/lib/child_process.js#L249) unnecessary? (The `code < 0` check is the only part of `lib/child_process.js` that doesn't currently get exercised in tests. It is run, but is always `false` so the `true` condition is untested.)